doc/guide/sonata-io.md

```diff
@@ -0,0 +1,174 @@
+- [mikroBUS][] - pin names have the form `mbN` where `N` is the physical pin number. The mikroBUS specification does not specify specific pin numbers. For Sonata `mb1` is mikroBUS pin `CS`, `mb4` is mikroBUS pin `MOSI`, `mb5` is mikroBUS pin `SDA` and `mb10` is mikroBUS pin `PWM`. The others follow from those. mikroBUS does not define any GPIO pins so there is no GPIO block for this header.
+
+## Peripheral Device Details
+
+The device names used here are those given the CHERIoT RTOS Sonata board description file found in [cheriot-rtos/sdk/boards/sonata-prerelease.json][].
+For each peripheral type details of the instances available and what pins they can mux to are given below along with a link to the driver file used for that peripheral.
+
+### SPI
+
+Driver: [cheriot-rtos/sdk/include/platform/platform-spi.hh][]
+
+2 of the SPI blocks are fixed.
+
+- `spi_lcd` - Connects to the LCD
+- `spi_ethmac` - Connects to the ethernet controller
+
+3 of the SPI instances are muxable.
+
+- `spi0` - SPI flash and SD card
+- `spi1` - RPI HAT SPI0, Arduino SPI, PMOD 0 SPI
+- `spi2` - RPI HAT SPI1, PMOD 1 SPI, Mikrobus SPI
+
+### UART
+
+Driver: [cheriot-rtos/sdk/include/platform/platform-uart.hh][]
+
+1 UART is fixed:
+
+- `uart` - The system UART available on the TX/RX0 UART header (P12).
+  Connects to FTDI USB UART when jumpers are connected.
+  Used as debug log, stdout and sterr in CHERIoT RTOS.
+
+The other 2 UARTs are muxable:
+
+- `uart1` - TX/RX1 UART header (P12), RPI Hat UART, Arduino UART, Mikrobus UART, PMOD 0 UART
+- `uart2` - TX/RX1 UART header (P12), PMOD1 UART, RS232, RS485
+
+### I2C
+
+Driver: [cheriot-rtos/sdk/include/platform/platform-i2c.hh][]
+
+Both I2C instances are muxable, however they act differently to the other muxable busses.
+As I2C is a low speed shared bus multiple headers can be muxed on to an I2C instance.
+So for each i2c blocks it's possible to have all the physical pins driven by it at once.
+
+- `i2c0` - qwiic0 & Arduino I2C, RPI Hat EEPROM I2C, Mikrobus I2C, PMOD0 I2C
+- `i2c1` - qwiic1, RPI Hat I2C, Mikrobus I2C, PMOD1 I2C
+
+Note that the qwiic0 & Arduino I2C pins are physically wired together on the PCB.
+
+### GPIO
+
+Driver: [cheriot-rtos/sdk/include/platform/platform-gpio.hh][]
+
+All of the GPIO blocks are fixed, in that each are dedicated to a specific set of pins.
+For many of those pins there's multiple things they can be muxed to (e.g. on the Raspberry Pi HAT header you can choose between using the SPI to drive the SPI pins or switch them to GPIO).
+
+The available GPIO blocks are:
+
+- `gpio_board` - LEDs and switches on the Sonata PCB
+- `gpio_rpi` - Raspberry PI HAT
+- `gpio_arduino` - Arduino Shfield
+- `gpio_pmod0` - PMOD 0
+- `gpio_pmod1` - PMOD 1
+- `gpio_pmodc` - PMOD C the middle 6 pins between PMOD 0 and PMOD 1
+
+### PWM
+
+Driver: [cheriot-rtos/sdk/include/platform/platform-pwm.hh][]
+
+One PWM block, with 7 channels, is muxable:
```

Comment:
Looks like a minor bug, we do in fact have 7 channels, however only 6 of them are connected to anything: https://github.com/lowRISC/sonata-system/blob/0ec0807fb740d8f14dd8a7cc2d8097bc56cbd1e0/data/top_config.toml#L18

I'll update this to 6 given the 7th isn't usable for anything.

Reply:
This is not a bug, more just a hard-coding of the LCD backlight PWM being defined in the Pinmux top config (so that it remains part of the same address region) but not actually being used by Pinmux: see [`sonata-system.sv`](https://github.com/lowRISC/sonata-system/blob/0ec0807fb740d8f14dd8a7cc2d8097bc56cbd1e0/rtl/system/sonata_system.sv#L942)

Reply:
Ah ok, I had assumed we had an entirely separate PWM block but it's one PWM block and the `pwm_lcd` device just points to the top set of registers?

Reply:
Yes, exactly (or the bottom set? Not sure what the convention for "top/bottom" is in this case but it points to the set of PWM registers with the largest start address in the address space).

I just made them separate devices in the board description file because it makes compartmentalisation & auditing nicer - you can give the LCD access to the PWM for its backlight without giving it access to all other PWMs.